### ISSUE TITLE
ControlArray Input Validation

### ISSUE THREAD

**Issue by ChristianBeilschmidt** [justindujardin/ng2-material#190]
Hi,

I have an input generated by ngFor:

```
<md-input-container class="md-block" *ngFor="#control of arrayControls.controls; #i = index">
    <label>Value for {{i}}</label>
    <input md-input [ngFormControl]="control">
    <div md-messages="???">
        <div md-message="required">You must specify a name.</div>
    </div>
</md-input-container>
```

```
let arrayControls = formBuilder.array([
    formBuilder.control("A", Validators.required),
    formBuilder.control("A", Validators.required)
]);
```

How do I link the md-messages with the validator?

Best regards

**Comment by steve-farr** [comment]
This works for me where the array of controls is the only thing in the form

```
<form [ngFormModel]="arrayControls">
  <md-input-container class="md-block" *ngFor="#ctrl of arrayControls.controls; #i = index">
    <label htmlFor="{{i}}">Value for {{i}}</label>
    <input md-input ngControl="{{i}}" id="{{i}}" #{{i}}="ngForm">
    <div md-messages="{{i}}">
      <div md-message="minlength">This must be at least 5 characters long.</div>
      <div md-message="maxlength">This must be no longer than 10 characters long.</div>
      <div md-message="required">You must specify a name</div>
    </div>
  </md-input-container>
</form>
```
with 
```
this.arrayControls = this.fb.array([
  fb.control("1234", Validators.compose([Validators.required, Validators.minLength(5), Validators.maxLength(10)])),
  fb.control("ABC", Validators.compose([Validators.required, Validators.minLength(5), Validators.maxLength(10)]))
]);
```

**Comment by justindujardin** [comment]
The input component has been replaced with the [Material2 input](https://github.com/angular/material2/tree/master/src/components/input) component as of version `0.4.1`.  Closing this issue.

**Comment by vincentpalita** [comment]
@justindujardin: well MdMessages still doesn't work with md-input and ControlArray.
Would be great to have this feature I guess. Will maybe try to work on this as soon as I have some time!

**Comment by vincentpalita** [comment]
@ChristianBeilschmidt : I made my way finding how to use it with md-input from @angular2/material and ng2-material/md-messages. FINALLY!

Inspired myself of the following plunkr: http://plnkr.co/edit/23DESOpbNnBpBHZt1BR4?p=preview

The keypoint to understand is that you **ALWAYS** need to nest fields inside the ControlGroup that you defined in your FormBuilder definition, **even** if it is a ControlArray.

To sum up, here is a plunkr I did about this very subject:
http://plnkr.co/edit/QX6mjoVNd7ZopsGzmKk7

I hope it helps!
